react/components/molecules/select/select.js

```diff
@@ -68,12 +72,8 @@ export class Select extends mix(PureComponent).with(IdentifiableMixin) {
             return;
         }
 
-        this.setState(
-            {
-                valueData: value
-            },
-            () => this.props.onUpdateValue(value)
-        );
+        this.props.onUpdateValue(value);
+        this.setState({ valueData: value });

```

Comment:
hum, shouldn't we only "send the event" once the value has indeed changed? Or was something happening that forced this change?

Reply:
That had a major performance impact for no real benefit. In fact we're just emitting the update before updating the component's internal state, I've seen different order in doing this and I can put `this.props.onUpdateValue(value);` after, as the most impactful change is not having to wait for the setState callback to emit that the value updated to the parent.

Reply:
Got it, nice! 👍